### PR TITLE
Parameter to exclude files and directories

### LINKED ISSUE #170: Requires parameters to exclude directories.
Please consider the following use-case:

```
assets/images/... -> should be generated
assets/data/... -> should not be generated
```

All files inside the data folder are simple data files, for example json or xml files. Which are used internally in the flutter application and therefore should not be considered as layout files (images, color, fonts etc.).

It would be great to have an `exclude: ` property to defines those files or directories.

_Originally posted by @jayjah in https://github.com/FlutterGen/flutter_gen/issues/132#issuecomment-986102743_

### PR DESCRIPTION
**Pull request by abelokon0711**
Fixes #170

### CODE REVIEW

File: packages/core/lib/generators/assets_generator.dart [hunk 1]
```diff
@@ -99,14 +102,20 @@ String generateAssets(
 List<String> _getAssetRelativePathList(
   String rootPath,
   List<String> assets,
+  List<String> excludes,
 ) {
   final assetRelativePathList = <String>[];
   for (final assetName in assets) {
     final assetAbsolutePath = join(rootPath, assetName);
+    if (excludes.any((exclude) => assetAbsolutePath.contains(exclude))) {
```
Comment (lcdsmao) [inline packages/core/lib/generators/assets_generator.dart:110]: How about making `exclude` to support the `glob` pattern which can be more convenient?
https://pub.dev/packages/glob

File: packages/core/lib/generators/assets_generator.dart [hunk 2]
```diff
@@ -99,14 +102,20 @@ String generateAssets(
 List<String> _getAssetRelativePathList(
   String rootPath,
   List<String> assets,
+  List<String> excludes,
 ) {
   final assetRelativePathList = <String>[];
```
Comment (lcdsmao) [inline packages/core/lib/generators/assets_generator.dart:107]: How about filtering the result of assetRelativePathList after the `for` expression.
Example:
```dart
final assetRelativePathList = <String>[];
for (final assetsName in assets) {
...
}
return assetRelativePathList.where(...).toList();
```

**Comment by tsinis** [comment]
Such a nice feature :(

**Review by AlexV525 — COMMENTED** [review]
Do we actually need the exclude config from the library? The build configurations already includes such feature.

**Comment by AlexV525** [comment]
Please submit a new request if this is still required.

**Pull request closed without merging**